plugin/path_static_account.go

```diff
@@ -0,0 +1,346 @@
+			},
+			"bindings": {
+				Type:        framework.TypeString,
+				Description: "Bindings configuration string.",
+			},
+			"token_scopes": {
+				Type:        framework.TypeCommaStringSlice,
+				Description: fmt.Sprintf(`List of OAuth scopes to assign to access tokens generated under this account. Ignored if "secret_type" is not "%q"`, SecretTypeAccessToken),
+			},
+		},
+		ExistenceCheck: b.pathStaticAccountExistenceCheck,
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.DeleteOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountDelete,
+			},
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountRead,
+			},
+			logical.CreateOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountCreate,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountUpdate,
+			},
+		},
+		HelpSynopsis:    pathStaticAccountHelpSyn,
+		HelpDescription: pathStaticAccountHelpDesc,
+	}
+}
+
+func pathStaticAccountList(b *backend) *framework.Path {
+	// Paths for listing role sets
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%ss?/?", staticAccountPathPrefix),
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ListOperation: &framework.PathOperation{
+				Callback: b.pathStaticAccountList,
+			},
+		},
+		HelpSynopsis:    pathListStaticAccountHelpSyn,
+		HelpDescription: pathListStaticAccountHelpDesc,
+	}
+}
+
+func (b *backend) pathStaticAccountExistenceCheck(ctx context.Context, req *logical.Request, d *framework.FieldData) (bool, error) {
+	nameRaw, ok := d.GetOk("name")
+	if !ok {
+		return false, errors.New("static account name is required")
+	}
+
+	acct, err := b.getStaticAccount(nameRaw.(string), ctx, req.Storage)
+	if err != nil {
+		return false, err
+	}
+
+	return acct != nil, nil
+}
+
+func (b *backend) pathStaticAccountRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	nameRaw, ok := d.GetOk("name")
+	if !ok {
+		return logical.ErrorResponse("name is required"), nil
+	}
+
+	acct, err := b.getStaticAccount(nameRaw.(string), ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if acct == nil {
+		return nil, nil
+	}
+
+	data := map[string]interface{}{
+		"service_account_project": acct.Project,
+		"service_account_email":   acct.EmailOrId,
+		"secret_type":             acct.SecretType,
+	}
+
+	if len(acct.Bindings) > 0 {
+		data["bindings"] = acct.Bindings.asOutput()
+	}
+	if acct.TokenGen != nil && acct.SecretType == SecretTypeAccessToken {
+		data["token_scopes"] = acct.TokenGen.Scopes
+	}
+
+	return &logical.Response{
+		Data: data,
+	}, nil
+}
+
+func (b *backend) pathStaticAccountDelete(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	nameRaw, ok := d.GetOk("name")
+	if !ok {
+		return logical.ErrorResponse("name is required"), nil
+	}
+	name := nameRaw.(string)
+
```

Comment:
I'm wondering if we should also take the `staticAccountLock` for deletion. A concurrent update to the static account could leave the service account with unexpected bindings or keys.

Reply:
Added in 51441cc